poc/mastic.py

```diff
@@ -108,74 +104,79 @@ def shard_without_joint_rand(
             measurement: tuple[tuple[bool, ...], W],
             nonce: bytes,
             rand: bytes,
-    ) -> tuple[MasticPublicShare, list[MasticInputShare]]:
+    ) -> tuple[list[CorrectionWord], list[MasticInputShare]]:
         (vidpf_rand, rand) = front(self.vidpf.RAND_SIZE, rand)
         (prove_rand_seed, rand) = front(self.xof.SEED_SIZE, rand)
         (helper_seed, rand) = front(self.xof.SEED_SIZE, rand)
         assert len(rand) == 0  # REMOVE ME
 
         (alpha, weight) = measurement
-        beta = self.flp.encode(weight)
+        beta = [self.field(1)] + self.flp.encode(weight)
 
         # Generate VIDPF keys.
         (correction_words, keys) = \
             self.vidpf.gen(alpha, beta, ctx, nonce, vidpf_rand)
 
         # Generate FLP and split it into shares.
         prove_rand = self.prove_rand(ctx, prove_rand_seed)
-        proof = self.flp.prove(beta, prove_rand, [])
+        proof = self.flp.prove(beta[1:], prove_rand, [])
```

Comment:
so this assumes the counter is always first, right? Could you leave a comment somewhere to make this more explicit?

Reply:
Done.